### ISSUE TITLE
Question about fine-tuning LLM-based retrievers

### ISSUE THREAD

**Issue by dkoterwa** [UKPLab/sentence-transformers#3460]
Hi!
Since LLM-based retrievers are becoming more and more popular, I was wondering if Sentence Transformers library has already introduced fine-tuning option for this kind of models. I mean fine-tuning based on EOS token. If the Pooling in my model is set to the EOS token pooling, then can I run the Trainer and it will work without any issues?

**Comment by tomaarsen** [comment]
Hello!

Yes, you can finetune decoder-style LLM models just like any other embedding model. When initializing the model, it is recommended to do so like this:
```python
from sentence_transformers import models, SentenceTransformer

transformer = models.Transformer("LiquidAI/LFM2-350M")
pooling = models.Pooling(transformer.get_word_embedding_dimension(), pooling_mode="lasttoken")
normalize = models.Normalize()  # Optional, this automatically adds embedding normalization such that dot product is equivalent to cosine similarity

model = SentenceTransformer(modules=[transformer, pooling, normalize])
```

This is near identical to the regular initialization with just `SentenceTransformer("LiquidAI/LFM2-350M")`, but with last-token (i.e. EOS) pooling. See the [Pooling](https://sbert.net/docs/package_reference/sentence_transformer/models.html#sentence_transformers.models.Pooling) module for more details, or the [Creating Custom Models](https://sbert.net/docs/sentence_transformer/usage/custom_models.html) documentation for more details on how to create Sentence Transformer models using several modules executed in sequence.

Do note that decoders are often finetuned using a strict chat template, whereas the Sentence Transformers training doesn't use this chat template - only the regular tokenization (e.g. BOS, EOS, padding tokens). This can mean a "slow start", or unexpectedly meh performance as the model tries to learn how to deal with texts that don't fully match the template used in decoder training. 

- Tom Aarsen

**Comment by onestardao** [comment]
This looks like it could be a case of Semantic ≠ Embedding — when the retrieved vector matches by distance but not by intended meaning (often caused by pooling or token handling choices). You might want to check if your EOS pooling is introducing a semantic shift.
